data/sql/updates/pending_db_world/rev_1555024514010587300.sql

```diff
@@ -0,0 +1,9 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1555024514010587300');
+
+SET @ENTRY := 21291;
+UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`= @ENTRY;
+DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(@ENTRY, 0, 0, 0, 64, 0, 100, 0, 0, 0, 0, 0, 54, 20000, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, "When player opened dialog - Give Pause path for 20000 ms"),
+(@ENTRY, 0, 1, 0, 11, 0, 100, 0, 0, 0, 0, 0, 53, 0, 21291, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, "On respawn  - Start path, walk, repeat, Passive"),
+(@ENTRY, 0, 2, 0, 1, 0, 100, 0, 10000, 30000, 240000, 240000, 80, 2129100, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, "When out of combat and timer at the begining - Start timed action list id #2129100");
```

Comment:
The entries referenced here by ID 1 and 2 already exist in the DB as 0 and 1. Why delete them? You could just add the new entry as ID 2:
```sql
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0 AND `id`=2;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(@ENTRY, 0, 2, 0, 64, 0, 100, 0, 0, 0, 0, 0, 54, 20000, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'When player opened dialog - Give Pause path for 20000 ms');
```
Also the update statement is not necessary, as the creature template already uses SmartAI. Please always check the DB if the data already exists. This could be potentially dangerous if deleting entries which are used otherwise (it is not the case here).

One last thing: Please use single quotes.

Reply:
> The entries referenced here by ID 1 and 2 already exist in the DB as 0 and 1. Why delete them? You could just add the new entry as ID 2:
> 
> ```sql
> DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0 AND `id`=2;
> INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
> (@ENTRY, 0, 2, 0, 64, 0, 100, 0, 0, 0, 0, 0, 54, 20000, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'When player opened dialog - Give Pause path for 20000 ms');
> ```
> 
> Also the update statement is not necessary, as the creature template already uses SmartAI. Please always check the DB if the data already exists. This could be potentially dangerous if deleting entries which are used otherwise (it is not the case here).
> 
> One last thing: Please use single quotes.

How i know it's always better to do delete before insert.  Yes, they re already exist ( Exist incorrect ) . Why i'm deleting 0,1, (2) doesn't exist. ( Now, They re totally incorrect, and as you can see i'm editing them to match correct Actions ).
As example i'm sure you didn't test this, and you didn't saw what's actually problem with this NPC. But i'll suggest you first to try steps included in PR then i'm able to receive suggestion :smiley: 
P.S Also, test it with this PR.
Grom'tor, Son of Oronok - On Respawn - Start Waypoint - Interact with him and check what's happening.
Grom'tor, Son of Oronok - Out of Combat - Run Script (No Repeat) - Keep continue 

Check the rows which exist already in AC db, and compate them with these included in PR

Reply:
I don't need to test this in order to compare the DB entries. Here's a dump without this PR:
```sql
INSERT INTO `smart_scripts` VALUES (21291,0,0,0,11,0,100,0,0,0,0,0,0,53,0,21291,1,0,0,0,1,0,0,0,0,0,0,0,0,'Grom\'tor, Son of Oronok - On Respawn - Start Waypoint (No Repeat)');
INSERT INTO `smart_scripts` VALUES (21291,0,1,0,1,0,100,0,10000,30000,240000,240000,0,80,2129100,0,0,0,0,0,1,0,0,0,0,0,0,0,0,'Grom\'tor, Son of Oronok - Out of Combat - Run Script (No Repeat)');
```

And these are the two entries you insert via this PR:
```sql
(@ENTRY, 0, 1, 0, 11, 0, 100, 0, 0, 0, 0, 0, 53, 0, 21291, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'On respawn  - Start path, walk, repeat, Passive'),
(@ENTRY, 0, 2, 0, 1, 0, 100, 0, 10000, 30000, 240000, 240000, 80, 2129100, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'When out of combat and timer at the begining - Start timed action list id #2129100');
```

They are clearly the same: same event types, same action types, same target types, same parameters. Only the comments are different, which is not important (I think the already existing comments are better). You are also not yet using the changed DB model. There's now an additional "event_param5" and "target_param4". Although it is not necessary to add these it would reflect the correct table columns.

So all in all it would suffice to just insert one single row to fix this, as I wrote above.

Reply:
> I don't need to test this in order to compare the DB entries. Here's a dump without this PR:
> 
> ```sql
> INSERT INTO `smart_scripts` VALUES (21291,0,0,0,11,0,100,0,0,0,0,0,0,53,0,21291,1,0,0,0,1,0,0,0,0,0,0,0,0,'Grom\'tor, Son of Oronok - On Respawn - Start Waypoint (No Repeat)');
> INSERT INTO `smart_scripts` VALUES (21291,0,1,0,1,0,100,0,10000,30000,240000,240000,0,80,2129100,0,0,0,0,0,1,0,0,0,0,0,0,0,0,'Grom\'tor, Son of Oronok - Out of Combat - Run Script (No Repeat)');
> ```
> 
> And these are the two entries you insert via this PR:
> 
> ```sql
> (@ENTRY, 0, 1, 0, 11, 0, 100, 0, 0, 0, 0, 0, 53, 0, 21291, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'On respawn  - Start path, walk, repeat, Passive'),
> (@ENTRY, 0, 2, 0, 1, 0, 100, 0, 10000, 30000, 240000, 240000, 80, 2129100, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'When out of combat and timer at the begining - Start timed action list id #2129100');
> ```
> 
> They are clearly the same: same event types, same action types, same target types, same parameters. Only the comments are different, which is not important (I think the already existing comments are better). You are also not yet using the changed DB model. There's now an additional "event_param5" and "target_param4". Although it is not necessary to add these it would reflect the correct table columns.
> 
> So all in all it would suffice to just insert one single row to fix this, as I wrote above.

They will be auto imported as 0 since they re empty, so that's not any problem.
![empty](https://user-images.githubusercontent.com/41213210/56038221-abe1e680-5d31-11e9-8d87-60bb9cd0f0b9.jpg)
Now take a look clearly on these 2 videos

Current:
[AC Current DB](https://youtu.be/N9TcwRP5HSs)

With this commit:
[AC with this PR](https://youtu.be/O8UQjEdh_PU)

Reply:
I'm not saying that this PR is not needed, I'm just saying that only 1 insert is necessary, you can test it yourself. Use a fresh DB and apply this:
```sql
DELETE FROM `smart_scripts` WHERE `entryorguid`=21291 AND `source_type`=0 AND `id`=2;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(21291, 0, 2, 0, 64, 0, 100, 0, 0, 0, 0, 0, 0, 54, 20000, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'When player opened dialog - Give Pause path for 20000 ms');
```
This will work, the other two entries don't need to be touched. I tested this myself, too.

And concerning "event_param5" and "target_param4", I know that they have 0 as default, but it is nethertheless good to see in the statement which values are entered.

Reply:
@Stoabrogga It is understood what you are trying to say. However I understand why is he doing this query. SAI Editor generates the complete registry for 1 Entry and since it hasn't been updated with param4 and param5, the query doesn't come with those.
We could say that he could manually add them to the query true but that would take much more time than creating the PR with the auto generated SQL. This is also the reason why queries come with double quotes instead of single quotes. 
Nevertheless I'm not saying that either is right or wrong. We do need however to have a tool that maximizes the Smart AI development speed and have it adapt to Azerothcore's standards.

We have been accepting Smart AI like this since very long time ago. Since we are slowly changing the mindset don't need to ask people to work completely differently of what they are used to. Baby steps 💪

Reply:
@Pondaveia 
Ok, I understand. I create the SAI scripts manually without tools, so it is easier for me to adapt to changes. The reasons why I do this are the same you mentioned: The tools are outdated or otherwise not suitable for AC. Had to check the source code more than once to understand what the different events and actions really do.

Reply:
@poszer 
Please add at least the name of the NPC to the comments, so it is documented which character this is all about:
```sql
(@ENTRY, 0, 0, 0, 64, 0, 100, 0, 0, 0, 0, 0, 54, 20000, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Grom''tor, Son of Oronok - When player opened dialog - Give Pause path for 20000 ms'),
(@ENTRY, 0, 1, 0, 11, 0, 100, 0, 0, 0, 0, 0, 53, 0, 21291, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Grom''tor, Son of Oronok - On respawn - Start path, walk, repeat, Passive'),
(@ENTRY, 0, 2, 0, 1, 0, 100, 0, 10000, 30000, 240000, 240000, 80, 2129100, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Grom''tor, Son of Oronok - When out of combat and timer at the beginning - Start timed action list id #2129100');
```
It would then be ok for me to merge as the other 2 entries which are re-inserted are basically the same as those which were already in the DB (only the comment will be different).

Reply:
> @poszer
> Please add at least the name of the NPC to the comments, so it is documented which character this is all about:
> 
> ```sql
> (@ENTRY, 0, 0, 0, 64, 0, 100, 0, 0, 0, 0, 0, 54, 20000, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Grom''tor, Son of Oronok - When player opened dialog - Give Pause path for 20000 ms'),
> (@ENTRY, 0, 1, 0, 11, 0, 100, 0, 0, 0, 0, 0, 53, 0, 21291, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Grom''tor, Son of Oronok - On respawn - Start path, walk, repeat, Passive'),
> (@ENTRY, 0, 2, 0, 1, 0, 100, 0, 10000, 30000, 240000, 240000, 80, 2129100, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Grom''tor, Son of Oronok - When out of combat and timer at the beginning - Start timed action list id #2129100');
> ```
> 
> It would then be ok for me to merge as the other 2 entries which are re-inserted are basically the same as those which were already in the DB (only the comment will be different).

Done